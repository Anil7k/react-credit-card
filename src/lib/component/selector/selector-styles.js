import { makeStyles } from '@material-ui/core';

export const useSelectorStyles = makeStyles(theme => ({
    margin: {
        marginRight: theme.spacing(1),
        minWidth: '100px',
    },
}));