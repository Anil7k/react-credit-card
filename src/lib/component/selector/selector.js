import React from "react";
import PropTypes from 'prop-types';
import {createTheme, ThemeProvider, withStyles} from "@material-ui/core/styles";
import {FormControl, InputBase, MenuItem, Select, Typography} from "@material-ui/core";
import {useSelectorStyles} from "./selector-styles";
import { StyledInput } from "../styles";

const theme = createTheme();

theme.typography.h6 = {
    fontSize: '12px',
    fontWeight: 600,
    color: '#1a3b5d',
    display: 'block',
};

export default function Selector(props) {
    const classes = useSelectorStyles(props);
    const { value, onChange, selectorList, placeholder, title, onCardFocus, onCardBlur} = props;

    return (
    <FormControl className={classes.margin}>
        <ThemeProvider theme={theme}>
            <Typography variant={'h6'}> {title ? title : '\u00A0' } </Typography>
        </ThemeProvider>
        <Select
            MenuProps={{
                anchorOrigin: {
                    vertical: "top",
                    horizontal: "left"
                },
                transformOrigin: {
                    vertical: "bottom",
                    horizontal: "left"
                },
                getContentAnchorEl: null
            }}
            labelId=""
            value={value}
            displayEmpty
            onChange={onChange}
            input={<StyledInput />}
            onFocus={(e) => onCardFocus(e, 'validity')}
            onBlur={onCardBlur}
        >
            <MenuItem value={''} disabled> { placeholder } </MenuItem>
            {selectorList.map((val, index) => (
                <MenuItem value={val}>{val}</MenuItem>
            ))}
        </Select>
    </FormControl>
)
}

Selector.propTypes = {
value: PropTypes.string,
onChange: PropTypes.func,
placeholder: PropTypes.string,
title: PropTypes.string,
selectorList: PropTypes.array,
};