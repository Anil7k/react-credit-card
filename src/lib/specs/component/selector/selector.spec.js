import React from 'react';
import { shallow } from 'enzyme';
import selector from '../../../component/selector/';

describe('selector', () => {
  it('render selector component', () => {
    const component = shallow(<selector debug />);

    expect(component).toMatchSnapshot();
  })
  it('should render selector with given params', () => {
    const component = shallow(<selector
        onChange={''}
        value={'2021'}
        selectorList={''}
        placeholder={'Year'}
        onCardFocus={''}
        onCardBlur={''}
    />);
    expect(component).toMatchSnapshot();
  });
});