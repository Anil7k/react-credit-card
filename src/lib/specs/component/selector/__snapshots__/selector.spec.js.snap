// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selector render selector component 1`] = `
<selector
  debug={true}
/>
`;

exports[`selector should render selector with given params 1`] = `
<selector
  onCardBlur=""
  onCardFocus=""
  onChange=""
  placeholder="Year"
  selectorList=""
  value="2021"
/>
`;
